### PR TITLE
set up database for saved connections, add test

### PR DESCRIPTION
**Pull request by larahynkim**

### CODE REVIEW

File: back-end/connections.js [hunk 1]
```diff
@@ -25,6 +28,22 @@ router.get('/saved-connections', async (req, res, next) => {
   }
 });
 
+router.get('/user-connections/:userId', async (req, res) => {
+  try {
+    const userId = req.params.userId;
+    console.log('User ID:', userId); 
+
+    const userConnections = await User.findById(userId).select('connections').populate('connections');
```
Comment (SivanC) [inline back-end/connections.js:36]: I don't see the need for populate here, but maybe I'm wrong

File: back-end/connections.js [hunk 2]
```diff
@@ -25,6 +28,22 @@ router.get('/saved-connections', async (req, res, next) => {
...
+    console.log('User Connections:', userConnections);     if (!userConnections) {
```
Comment (SivanC) [inline back-end/connections.js:37]: syntax error? removing your semicolons would help to catch these

File: front-end/src/SavedConnections.js [hunk 3]
```diff
@@ -9,8 +10,20 @@ const SavedConnections = () => {
...
+        if (!token) {
+          console.error('No token found');
+          return;
+        }
+
+        // Decode token to get user ID
+        const decoded = jwtDecode(token); 
+        const userId = decoded.userId;
+
+        const apiUrl = `${process.env.REACT_APP_API_URL_LK}/user-connections/${userId}`;
```
Comment (SivanC) [inline front-end/src/SavedConnections.js:23]: `api/saved-connections/user-connections` is right? Seems a little long

File: front-end/src/SavedConnections.js [hunk 4]
```diff
@@ -19,15 +32,15 @@ const SavedConnections = () => {
...
+  
   return (
     <div className="saved-connections-container">
       <div className="header">
       <h2>Saved Connections</h2>
       </div>
       <div className="connections-grid">
         {connections.map((connection) => (
-          <Link key={connection.id} to={`/ConnectionDetails/${connection.id}`} className="connection-box">
+          <Link key={connection.id} to={`/ConnectionDetails/${connection._id}`} className="connection-box">
```
Comment (SivanC) [inline front-end/src/SavedConnections.js:43]: `.id` is a valid way to get the id

**Review by SivanC — COMMENTED** [review]
Code looks pretty good, can't launch front end because you didn't make jwt decode a dependency on front-end. Each directory (back/front) has their own package.json, so make sure you're in the right directory when adding dependencies

File: back-end/authRoutes.js [hunk 5]
```diff
@@ -5,10 +5,25 @@ const router = express.Router();
...
+  };*/
   
+const generateToken = (userId) => {
+    return jwt.sign({ userId }, 'secretKey', { expiresIn: '24h' }); 
+};
+
+
+const authenticateToken = (req, res, next) => {
+  const token = req.headers['authorization'];
+  if (token == null) return res.sendStatus(401);
```
[the remark this answers is not in the document]
Comment (larahynkim) [reply back-end/authRoutes.js:19]: i used == null as a concise way to check for both null and undefined!

File: back-end/connections.js [hunk 6]
```diff
@@ -2,11 +2,14 @@ const express = require('express');
...
+const mongoose = require('mongoose');
+const User = require('./models/User.js');
 
 const apiBaseUrl = process.env.API_BASE_URL;
 const apiKey = process.env.LARA_API_KEY;
+const authenticateToken = require('./authRoutes');
 
-router.get('/saved-connections', async (req, res, next) => {
+router.get('/saved-connections', authenticateToken, async (req, res, next) => {
   if (process.env.SIMULATE_ERROR === 'true') {
```
[the remark this answers is not in the document]
Comment (larahynkim) [reply back-end/connections.js:13]: this is for the tests! i'm simulating an error to test for error handling in the tests

[the remark this answers is not in the document]
Comment (SivanC) [reply back-end/connections.js:14]: At this point, you should be able to properly generate an error by sending a request to the server, and not have to fake it. Faking it only tests that you can fake it, you get what I mean?

File: back-end/models/User.js [hunk 7]
```diff
@@ -1,4 +1,5 @@
 const mongoose = require('mongoose');
+const { ObjectId } = mongoose.Schema.Types;
```
Comment (SivanC) [inline back-end/models/User.js:2]: realized I forgot to add this in my own copy of the schema, changed it to `mongoose.ObjectId` on the line itself, up to you which is better when you merge

**Review by SivanC — APPROVED** [review]

File: back-end/connections.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (larahynkim) [reply back-end/connections.js:14]: Ah, I understand now. I will work on that after the other routes are set up!

**Pull request merged**